File: paack-go-sdk/rest/retailer_location_api_test.go
package rest

import (
	"github.com/PaackEng/paack-go-sdk/types"
	"github.com/stretchr/testify/assert"
	"testing"
)

var retailerLocationApi *RetailerLocationApi

func RetailerLocationApiInit() *RetailerLocationApi {
	if retailerLocationApi != nil {
		return retailerLocationApi
	}

	apiClient, config, customLogger, _, retailerAudience := BaseInit()
	retailerLocationApi = NewRetailerLocationApi(retailerAudience, config.Result.DataConfig.Domain.Coverage.Staging, &config.Result.DataConfig.Resources, *apiClient, customLogger)

	return retailerLocationApi
}

var retailerLocationId = "008f3945-7c38-4569-8704-62702b8487d0"

var retailerLocation = types.RetailerLocation{
	Address: types.Address{
		City:     "Barcelona",
		Country:  "ES",
		County:   "Comtat de Barcelona",
		Line1:    "Via Augusta",
		Line2:    "17, Principal",
		PostCode: "08006",
	},
	Alias:        "TestingRL28123",
	DcID:         "dc149383-2071-4ea1-95de-57603ef47d66",
	RetailerID:   "ab4ce553-ad39-4cfa-ba81-0f4b9b22efcf",
	LocationName: "Testing Retailer28123",
	RetailerName: "Testing Retailer28",
	Type:         "Store",
}

func Test_CreateRetailerLocation(t *testing.T) {

	RetailerLocationApiInit()

	got, err := retailerLocationApi.CreateRetailerLocation(retailerLocation)

	assert.NotNil(t, got)
	assert.Nil(t, err)

	if err == nil {
		retailerLocationId = got.Success.RetailerLocationID
	}
}

func Test_GetRetailerLocation(t *testing.T) {

	RetailerLocationApiInit()

	request := GetRetailerLocationRequest{
		//ID: "71c375e2-eccf-48e5-813f-6d0b9d756f26",
		RetailerID: "ab4ce553-ad39-4cfa-ba81-0f4b9b22efcf",
		//RetailerName: "Testing Retailer28",
		//	LocationName: "Testing Retailer28",
		//	Country:      "ES",
		//	City:         "Barcelona",
		//	PostCode:     "08006",
		//	DcID:         "dc149383-2071-4ea1-95de-57603ef47d66",
		//	Alias:        "TestingRL28",
		//Type: "Store",
	}

	got, err := retailerLocationApi.GetRetailerLocation(request)

	assert.NotNil(t, got)
	assert.Nil(t, err)
}

var patchRetailerLocation = PatchRetailerLocationRequest{
	Address: &types.Address{
		City:     "Barcelona2",
		Country:  "ES",
		County:   "Comtat de Barcelona2",
		Line1:    "Via Augusta2",
		Line2:    "17, Principal2",
		PostCode: "08006",
	},
	Alias: "TestingRL282",
	DcID:  "dc149383-2071-4ea1-95de-57603ef47d66",
	Type:  "Store",
}

func Test_UpdateRetailerLocation(t *testing.T) {

	RetailerLocationApiInit()

	got, err := retailerLocationApi.UpdateRetailerLocation(retailerLocationId, patchRetailerLocation)

	assert.NotNil(t, got)
	assert.Nil(t, err)
}

func Test_DeleteRetailerLocation(t *testing.T) {

	RetailerLocationApiInit()

	_, err := retailerLocationApi.DeleteRetailerLocation(retailerLocationId)

	assert.Nil(t, err)
}

File: paack-java-sdk/src/test/java/eu/paack/sdk/api/RetailerLocationAPITest.java
package eu.paack.sdk.api;

import eu.paack.sdk.Paack;
import eu.paack.sdk.api.model.request.RetailerGetRequest;
import eu.paack.sdk.api.model.response.Error;
import eu.paack.sdk.api.model.response.PaackResponse;
import eu.paack.sdk.api.model.response.RetailerLocationResponse;
import eu.paack.sdk.config.Domain;
import eu.paack.sdk.model.Country;
import eu.paack.sdk.model.RetailerLocationAddress;
import eu.paack.sdk.model.RetailerLocation;
import org.junit.Before;
import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertNull;

public class RetailerLocationAPITest {

    Paack paack;

    @Before
    public void setUp() {
        String clientId = System.getEnv("CLIENT_ID");
        String clientSecret = System.getEnv("CLIENT_SECRET");

        paack = new Paack(clientId, clientSecret, Domain.STAGING);
    }

    @Test
    public void testCreate() {
        RetailerLocationAddress address = RetailerLocationAddress.builder()
                .city("Barcelona")
                .country(Country.SPAIN.getCountryCode())
                .county("Comtat de Barcelona")
                .line1("Via Augusta 88")
                .line2("17, Principal")
                .post_code("08006")
                .build();

        RetailerLocation retailerRequest = RetailerLocation.builder()
                .address(address)
                .alias("TestingRL281240")
                .retailerId("866b263f-ff80-47e8-83d8-56345f151e6f")
                .locationName("Testing Retailer281241")
                .retailerName("Testing Retailer31")
                .type("Store")
                .build();

        PaackResponse<RetailerLocationResponse, Error> response = paack.retailerLocation().create(retailerRequest);

        assertNull(response.getError());

        paack.retailerLocation().delete(response.getData().getRetailerLocationId());
    }

    @Test
    public void testDelete() {
        RetailerLocationAddress address = RetailerLocationAddress.builder()
                .city("Barcelona")
                .country(Country.SPAIN.getCountryCode())
                .county("Comtat de Barcelona")
                .line1("Via Augusta 88")
                .line2("17, Principal")
                .post_code("08006")
                .build();

        RetailerLocation retailerRequest = RetailerLocation.builder()
                .address(address)
                .alias("TestingRL281240")
                .retailerId("866b263f-ff80-47e8-83d8-56345f151e6f")
                .locationName("Testing Retailer281241")
                .retailerName("Testing Retailer31")
                .type("Store")
                .build();

        PaackResponse<RetailerLocation, Error> response = paack.retailerLocation()
                .delete("ab4ce553-ad39-4cfa-ba84-0f4b9b22efcf");

        assertNull(response.getError());
    }

    @Test
    public void testGet() {
        RetailerGetRequest retailerGetRequest = RetailerGetRequest.builder().alias("TestingRL281235").build();

        PaackResponse<List<RetailerLocation>, Error> response = paack.retailerLocation()
                .get(retailerGetRequest);

        assertNull(response.getError());
    }
}
